### PR TITLE
Better configuration for front end

### PR DESCRIPTION
**Pull request by k88hudson**
OK, so this does a few things:

* allows for a local settings.conf.js file that's gitignored
* groups all configuration under one global (`__CONFIG__`)
* uses a common file to generate webpack settings for karma/webpack

I wrote the node stuff in node 4 so I need to update travis to use that

### CODE REVIEW

**Comment by tkiethanom** [comment]
Cool! Moving the settings file out of the repository is much better. Glad you're making these things more efficient.

**Comment by k88hudson** [comment]
@tkiethanom thanks! what do you think about doing 

```
config.get('FOO')
```
v.s.
```
config.FOO
``

The first style is kinda safer for preventing silent errors because it throws if `FOO` is undefined, but requires a bit of boilerplate

**Comment by tkiethanom** [comment]
@k88hudson Safer sounds better to me, even if there is a little boilerplate.

**Comment by k88hudson** [comment]
Ok all fixed up and rebased. r? @tkiethanom @rlr?

File: front_end/src/actions/AdGroups/AdGroupActions.js [hunk 1]
```diff
@@ -1,11 +1,7 @@
 import fetch from 'isomorphic-fetch';
+import * as config from 'helpers/config';
 
-let apiUrl;
-if (typeof __DEVELOPMENT__ !== 'undefined' && __DEVELOPMENT__ === true) {
-  apiUrl = __DEVAPI__;
-} else {
-  apiUrl = __LIVEAPI__;
-}
+const apiUrl = config.get('DEVELOPMENT') ? config.get('DEV_API') : config.get('LIVE_API');
```
Comment (rlr) [inline front_end/src/actions/AdGroups/AdGroupActions.js]: Now that we have nicer settings. Should we look at just having one `API_URL` setting? Having only DEV_ and LIVE_ doesn't really help given we have local, dev, stage and prod environments.

Doesn't have to go in this PR.

**Comment by rlr** [comment]
I like! r+ from me

File: front_end/src/actions/AdGroups/AdGroupActions.js (hunk 1 above)
Comment (k88hudson) [inline front_end/src/actions/AdGroups/AdGroupActions.js]: Yeah i agree

**Comment by k88hudson** [comment]
Actually, this might affect our ops situation depending on how we're configuring stuff at the moment and also what version of node we're running, let me just check on that before merging

**Comment by k88hudson** [comment]
@rlr ok i fixed it up so it will work for production as well

**Comment by rlr** [comment]
thank you! r+

**Pull request merged**